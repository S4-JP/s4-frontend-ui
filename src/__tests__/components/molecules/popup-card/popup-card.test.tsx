import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { PopupCard } from '../../../../index'

const Dummy = () => <span>dummy</span>
const Ref = jest.fn()

afterEach(() => Ref.mockReset())

describe('ポップアップカードのレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    render(
      <PopupCard isShow={true} cardRef={Ref}>
        <Dummy />
      </PopupCard>
    )
    expect(screen.getByTestId('molecules-pc-wrap')).toMatchSnapshot()
    expect(screen.getByTestId('molecules-pc-wrap')).toHaveStyle('width: 320px;')
  })

  test('propsに設定した値でレンダリングされる。', () => {
    render(
      <PopupCard
        title="test title"
        width={100}
        isShow={true}
        cardRef={Ref}
        className="test"
      >
        <Dummy />
      </PopupCard>
    )
    expect(screen.getByTestId('molecules-pc-wrap')).toHaveAttribute(
      'width',
      '100'
    )
    expect(screen.getByTestId('molecules-pc-title').textContent).toBe(
      'test title'
    )
    expect(screen.getByTestId('molecules-pc-content')).toContainHTML(
      '<span>dummy</span>'
    )

    fireEvent.click(screen.getByTestId('molecules-pc-wrap'))
    expect(Ref).toHaveBeenCalledTimes(1)
  })
})
