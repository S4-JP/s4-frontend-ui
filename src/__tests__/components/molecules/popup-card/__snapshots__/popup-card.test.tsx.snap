// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ポップアップカードのレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-kTvvXX kbzdem"
  data-testid="molecules-pc-wrap"
>
  <div
    class="sc-jtcaXd eQQKuv"
    data-testid="molecules-pc-content"
  >
    <span>
      dummy
    </span>
  </div>
</div>
`;
